matrix_client/room.py

```diff
@@ -54,6 +56,32 @@ def set_user_profile(self,
             }
         )
 
+    @property
+    def display_name(self):
+        """
+        Calculates the display name for a room.
+        """
+        if self.name:
+            return self.name
+        elif self.canonical_alias:
+            return self.canonical_alias
+
+        members = self.get_joined_members()
```

Comment:
Since this is a property, it's reasonable to assume that it will be called
relatively frequently. `get_joined_members` needs changed so that it caches the
result, and this property doesn't result in an api call each time it's accessed.

Reply:
i've added `cachetools` to get `lru_cache` for `py2`, `py3`.

Reply:
Removed `cachetools` and use pure man `if exists use` now we can also test it with `_mkmembers`